sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -314,19 +322,29 @@ abstract class UnaryNode extends LogicalPlan {
    * expressions with the corresponding alias
    */
   protected def getAliasedConstraints(projectList: Seq[NamedExpression]): Set[Expression] = {
-    var allConstraints = child.constraints.asInstanceOf[Set[Expression]]
-    projectList.foreach {
-      case a @ Alias(e, _) =>
-        // For every alias in `projectList`, replace the reference in constraints by its attribute.
-        allConstraints ++= allConstraints.map(_ transform {
-          case expr: Expression if expr.semanticEquals(e) =>
-            a.toAttribute
-        })
-        allConstraints += EqualNullSafe(e, a.toAttribute)
-      case _ => // Don't change.
-    }
-
-    allConstraints -- child.constraints
+    val relativeReferences = AttributeSet(projectList.collect {
+      case a: Alias => a
+    }.flatMap(_.references))
+    val parAllConstraints = child.constraints.asInstanceOf[Set[Expression]].filter { constraint =>
+      constraint.references.intersect(relativeReferences).nonEmpty
+    }.par
+    parAllConstraints.tasksupport = UnaryNode.taskSupport
```

Comment:
Why are we using a custom task support instead of the default (which uses the global fork-join executor)?